daemon/daemon.go

```diff
@@ -744,6 +745,7 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 }
 
 func (daemon *Daemon) Shutdown() error {
+	daemon.shutdown = true
```

Comment:
I'm cool with this, we do stop taking API requests when the daemon starts shutting down.

@abronan @calavera are we good?